### ISSUE TITLE
Question about implementation of sinusoidal positional embedding

### ISSUE THREAD

**Issue by rlawjdghek** [facebookresearch/fairseq#4648]
Is there any reason why not shuffling sin and cos values when calculating positional embedding?

Because to the best of my knowledge, "attention is all you need" used shuffled sinusoidal positional embedding values. 

e.g.) sin, cos, sin, cos ...

But I see the implementation be like: torch.cat([sin,cos], dim=-1), which would be like: sin, sin, sin, ..., cos, cos, cos ...

I would appreciate if you let me know. THanks!
